odxtools/diaglayerraw.py

```diff
@@ -62,8 +62,8 @@ class DiagLayerRaw(IdentifiableElement):
     parent_refs: List[ParentRef]
     comparams: List[ComparamInstance]
     ecu_variant_patterns: List[EcuVariantPattern]
-    # comparam_spec: OdxLinkRef # TODO
-    # prot_stack_snref: str # TODO
+    comparam_spec_ref: OdxLinkRef
+    prot_stack_snref: str
```

Comment:
These should be optional because there are diag layers which don't exhibit these (e.g., `ECU-SHARED-DATA`):
```suggestion
    comparam_spec_ref: Optional[OdxLinkRef]
    prot_stack_snref: Optional[str]
```

Again, the "correct" solution is to split this class like in the specification. This is quite a bit of work though, because there are so many of them: `DiagLayer`, `HierarchyElement`, `EcuSharedData`, `EcuVariant`, ...